shared-module/displayio/Sprite.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Scott Shawcroft for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "shared-bindings/displayio/Sprite.h"
+
+
+#include "shared-bindings/displayio/Bitmap.h"
+
+void common_hal_displayio_sprite_construct(displayio_sprite_t *self, mp_obj_t bitmap,
+        mp_obj_t palette, uint16_t width, uint16_t height, uint16_t x, uint16_t y) {
+    self->width = width;
+    self->height = height;
+    self->bitmap = bitmap;
+    self->palette = palette;
+    self->x = x;
+    self->y = y;
+}
+
+void common_hal_displayio_sprite_get_position(displayio_sprite_t *self, int16_t* x, int16_t* y) {
+    *x = self->x;
+    *y = self->y;
+}
```

Comment:
I wounder if it would make sense to use a union of a 32-bit integer and two 16-bit integers to store the position. Then you could simply pass it around.

Reply:
Ya, maybe. It's trading a second load or store for a shift or mask. We can experiment as we optimize.

Reply:
I was more thinking about convenience/clarity of writing the code, than optimization, to be honest.

Reply:
Agree re clarity.